### PR TITLE
Text Search and Filters don't work at the same time in the web UI #1316

### PR DESCRIPTION
**Pull request by neel-suthar**
Making sure to return distinct results. Changing set operation to '&' to show the matching results from filters AND search term

Fixes: https://github.com/ArchiveBox/ArchiveBox/issues/1316

# Summary

The fix involves adjusting the combination of filter and search results to avoid duplicates. Instead of using the "or" (|) operation, consider using a method that ensures the snapshots appearing in the result set satisfy both the filter conditions and the search term. This adjustment aims to eliminate duplicate entries in the combined result.

Before the fix:
image

After the fix:
image

# Changes these areas

- [x] Bugfixes
- [ ] Feature behavior
- [ ] Command line interface
- [ ] Configuration options
- [ ] Internal architecture
- [ ] Snapshot data layout on disk

### CODE REVIEW

**Comment by neel-suthar** [comment]
@pirate Please review.

**Comment by pirate** [comment]
I'm worried that while this enables filters working, it will hide some results that only match via the search index but dont match via db field search.

e.g. if user searches for some string that's only present in a tag, but not in the snapshot content

Maybe better to extract the filter parameters passed from the UI and apply them to the queryset manually?

**Comment by neel-suthar** [comment]
Interesting. I thought the search results would contain that information. So when we do perform & operation the final set would have the results matching both. Let me go through the code and figure it out. Thanks.

File: archivebox/core/mixins.py [hunk 1]
```diff
@@ -10,10 +10,10 @@ def get_search_results(self, request, queryset, search_term: str):
 
         search_term = search_term.strip()
         if not search_term:
-            return qs, use_distinct
+            return qs.distinct(), use_distinct
         try:
             qsearch = query_search_index(search_term)
-            qs = qs | qsearch
+            qs = qs & qsearch
```
Comment (pirate) [inline archivebox/core/mixins.py:16]:
```suggestion
            qs = qs | qsearch
```

reverting just this change for now so I can merge the `.distinct()` fix

**Pull request merged**